Gems/Atom/RHI/Code/Source/RHI/MultiDeviceSwapChain.cpp

```diff
@@ -0,0 +1,283 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#include <Atom/RHI/Factory.h>
+#include <Atom/RHI/MemoryStatisticsBuilder.h>
+#include <Atom/RHI/MultiDeviceSwapChain.h>
+#include <Atom/RHI/RHISystemInterface.h>
+
+namespace AZ::RHI
+{
+    bool MultiDeviceSwapChain::ValidateDescriptor(const SwapChainDescriptor& descriptor) const
+    {
+        if (Validation::IsEnabled())
+        {
+            const bool isValidDescriptor = descriptor.m_dimensions.m_imageWidth != 0 && descriptor.m_dimensions.m_imageHeight != 0 &&
+                descriptor.m_dimensions.m_imageCount != 0;
+
+            if (!isValidDescriptor)
+            {
+                AZ_Warning("MultiDeviceSwapChain", false, "MultiDeviceSwapChain display dimensions cannot be 0.");
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    ResultCode MultiDeviceSwapChain::Init(MultiDevice::DeviceMask deviceMask, const SwapChainDescriptor& descriptor)
+    {
+        if (!ValidateDescriptor(descriptor))
+        {
+            return ResultCode::InvalidArgument;
+        }
+
+        if (descriptor.m_isXrSwapChain)
+        {
+            m_xrSystem = RHI::RHISystemInterface::Get()->GetXRSystem();
+            AZ_Assert(m_xrSystem, "XR System is null");
+        }
+
+        SwapChainDimensions nativeDimensions = descriptor.m_dimensions;
+        ResultCode resultCode = MultiDeviceResourcePool::Init(
+            deviceMask,
+            [this, &descriptor, &nativeDimensions]()
+            {
+                ResultCode result = ResultCode::Success;
+
+                IterateDevices(
+                    [this, &descriptor, &result, &nativeDimensions](int deviceIndex)
+                    {
+                        auto* device = RHISystemInterface::Get()->GetDevice(deviceIndex);
+
+                        m_deviceSwapChains[deviceIndex] = Factory::Get().CreateSwapChain();
+                        result = m_deviceSwapChains[deviceIndex]->Init(*device, descriptor);
+                        nativeDimensions = m_deviceSwapChains[deviceIndex]->GetDescriptor().m_dimensions;
```

Comment:
This will be the dimension returned by the last device. Hope that doesn't cause issues.